pkg/models/datasource_cache.go

```diff
@@ -84,19 +84,22 @@ func (ds *DataSource) GetHTTPTransport(provider httpclient.Provider, customMiddl
 
 func (ds *DataSource) HTTPClientOptions() sdkhttpclient.Options {
 	tlsOptions := ds.TLSOptions()
+	timeouts := &sdkhttpclient.TimeoutOptions{
+		Timeout:               ds.getTimeout(),
+		DialTimeout:           time.Duration(setting.DataProxyDialTimeout) * time.Second,
+		KeepAlive:             time.Duration(setting.DataProxyKeepAlive) * time.Second,
+		TLSHandshakeTimeout:   time.Duration(setting.DataProxyTLSHandshakeTimeout) * time.Second,
+		ExpectContinueTimeout: time.Duration(setting.DataProxyExpectContinueTimeout) * time.Second,
+		MaxConnsPerHost:       setting.DataProxyMaxConnsPerHost,
+		MaxIdleConns:          setting.DataProxyMaxIdleConns,
```

Comment:
```suggestion
		MaxIdleConnsPerHost: setting.DataProxyMaxIdleConns,
```
we still need to set MaxIdleConnsPerHost since the default is too low. But we can set it to the same as `MaxIdleConns` since the max count is per datasource.

Reply:
Oh, okay, I thought that we don't need that anymore as per the above comments. Fixed, thanks!